pkg/api/kubegraph/edges.go

```diff
@@ -0,0 +1,46 @@
+package kubegraph
+
+import (
+	"github.com/gonum/graph"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	kubegraph "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	ExposedThroughServiceEdgeKind = "ExposedThroughService"
+)
+
+// AddFullfillingDeploymentConfigEdges ensures that a directed edge exists between all deployment configs and the
+// services that expose them (via label selectors).
+func AddFullfillingDeploymentConfigEdges(g osgraph.MutableUniqueGraph, node *kubegraph.ServiceNode) *kubegraph.ServiceNode {
```

Comment:
This merely keeps the link in the existing direction.  No matter which way you do this, it will be O(n^2) overall (for all services: for all podtemplatespecs)